### PR TITLE
Add ci.yml to run unit tests and example DAGs in CI

### PR DESCRIPTION
**Pull request by pankajkoti**
Add ci.yml to run unit tests and example DAGs present in the tests directory.

### CODE REVIEW

File: cosmos/providers/databricks/constants.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
 import os
 
 JOBS_API_VERSION = os.getenv("JOBS_API_VERSION", "2.1")
+TEST = "test"
```
Comment (pankajkoti) [inline cosmos/providers/databricks/constants.py:4]:
```suggestion
```

File: pyproject.toml [hunk 2]
```diff
@@ -47,7 +47,7 @@ dbt-all = [
     "dbt-postgres",
     "dbt-redshift",
     "dbt-snowflake",
-    "dbt-openlineage"
+    "astronomer-cosmos[dbt-openlineage]"
```
Comment (pankajkoti) [inline pyproject.toml:50]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+env:
+  AIRFLOW__CORE__ALLOWED_DESERIALIZATION_CLASSES: "airflow.* astro.* cosmos.*"
+  CI_ENABLED: true
+
+jobs:
+# TODO: Fix Markdown-link-check failures
```
Comment (pankajkoti) [inline .github/workflows/ci.yml:29]:
```suggestion
# TODO: Fix Markdown-link-check failures https://github.com/astronomer/astronomer-cosmos/issues/172
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+#  Markdown-link-check:
+#    if: github.event.action != 'labeled'
+#    runs-on: ubuntu-latest
+#    steps:
+#      - uses: actions/checkout@v3
+#      - uses: gaurav-nelson/github-action-markdown-link-check@v1
+#        with:
+#          config-file: '.github/workflows/mlc_config.json'
+#
+# TODO: Fix Type-check failures
```
Comment (pankajkoti) [inline .github/workflows/ci.yml:39]:
```suggestion
# TODO: Fix Type-check failures https://github.com/astronomer/astronomer-cosmos/issues/173
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+env:
+  AIRFLOW__CORE__ALLOWED_DESERIALIZATION_CLASSES: "airflow.* astro.* cosmos.*"
+  CI_ENABLED: true
+
+jobs:
+# TODO: Fix Markdown-link-check failures https://github.com/astronomer/astronomer-cosmos/issues/172
```
Comment (tatiana) [inline .github/workflows/ci.yml:29]: To be addressed: https://github.com/astronomer/astronomer-cosmos/issues/172

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+#  Markdown-link-check:
+#    if: github.event.action != 'labeled'
+#    runs-on: ubuntu-latest
+#    steps:
+#      - uses: actions/checkout@v3
+#      - uses: gaurav-nelson/github-action-markdown-link-check@v1
+#        with:
+#          config-file: '.github/workflows/mlc_config.json'
+#
+# TODO: Fix Type-check failures https://github.com/astronomer/astronomer-cosmos/issues/173
```
Comment (tatiana) [inline .github/workflows/ci.yml:39]: To be addressed: https://github.com/astronomer/astronomer-cosmos/issues/173

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+      - uses: actions/setup-python@v3
+        with:
+          python-version: ${{ matrix.version }}
+          architecture: 'x64'
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/pip
+            .nox
+          key: tests-os-${{ runner.os }}-python-${{ matrix.python }}-airflow-${{ matrix.airflow }}-deps-${{ hashFiles('pyproject.toml') }}-version-${{ hashFiles('cosmos/__init__.py') }}
```
Comment (tatiana) [inline .github/workflows/ci.yml:118]: @pankajkoti and I checked which version customer B is using, and they are using Airflow 2.2.4.

We agreed to automate the checks (tests & example DAGs) for this version of Airflow as well in a separate ticket:
https://github.com/astronomer/astronomer-cosmos/issues/174

**Review by tatiana — APPROVED** [review]

**Pull request merged**